### PR TITLE
Fix #1200 show errors on source files with paths lengths over 255 chars

### LINKED ISSUE #1200: Compiling Errors not highligthted for source files having long paths (longer than the record length of the events file)
**Bug Description**

Source files that has errors and has a long path whose length is longer than the record length of the events file, when compile it the extension is not capable of highlighting it over the source code tab. If you go to problems tab you can see the messages but if any of this is clicked and an error is produced.

**Steps to reproduce the behavior**

1. Create a local project structure with many directories with large names or a big one using the max length allowed. This can be done working directly under the IFS instead of create a project.
2. Create a file with a long name inside the directory.
3. Deploy and compile.

**Expected behavior**
Errors highligthted with squiggly lines showed in the source file tab that was compiled.

**Screenshots**

image

image

image

**Environment**
 - Extension version 1.7.11
 - IBM i OS version 7.5

**Additional context**

The error parsers that process the messages from the events file are not taking care of the FILEIDCONT record type.

### PR DESCRIPTION
**Pull request by angelorpa**
### Changes

Add functions to work with lines of the events file and get the real path of the source file that was compiled taking care of files with paths lengths more than 255 characters to fix #1200 

### Checklist

* [x] have tested my change
* [x] updated relevant documentation
* [x] Remove any/all `console.log`s I added
* [x] eslint is not complaining
* [x] have added myself to the contributors' list in [CONTRIBUTING.md](https://github.com/halcyon-tech/vscode-ibmi/blob/master/CONTRIBUTING.md)
* [x] **for feature PRs**: PR only includes one feature enhancement.

### CODE REVIEW

File: src/api/errors/handler.ts
Comment (worksofliam) [inline src/api/errors/handler.ts:1]: Please run VS Code's formatter on this file.

1. Open the command palette (F1 / Cntrl+Shift+P)
2. Search 'Format Document'

**Review by worksofliam — CHANGES_REQUESTED** [review]
Works as expected - thanks for a great change.

Before I merge, please see my comment about formatting and then we should be good to go.

Liam

File: src/api/errors/handler.ts
Comment (angelorpa) [reply src/api/errors/handler.ts:1]: hi @worksofliam I ran the 'Format Document' but nothing change over the file, so i think this is according to the default formatting style.

Comment (angelorpa) [reply src/api/errors/handler.ts:1]: May be I need to set some config or style rules?, please let me know to take it present for nexts contributions.

Comment (angelorpa) [reply src/api/errors/handler.ts:1]: #[1d584e6](https://github.com/halcyon-tech/vscode-ibmi/pull/1276/commits/1d584e6555c6424c4c9fb5636c971c2a7fee9ee6)

@worksofliam  I wasn't using the default formatter, i think this is changes that you are specting.  check it and let me know.

**Comment by worksofliam** [comment]
Functionally works good and looks like the code cleanup is good too!

@chrjorgensen @sebjulliand This is a pretty impact change. Any chance you  two can have a play with compiling code to ensure errors continue to show as they did before?

I plan on writing some tests for this code in another PR.

**Review by sebjulliand — COMMENTED** [review]
I played around with it and everything worked fine. I compiled from a local project, a remote PF and a remote streamfile (with a long name) and errors were correctly reported.

Great job; code looks good too! 😃

**Comment by worksofliam** [comment]
@angelorpa Before we can merge this, we need you to pull in the latest changes from `master` and add a test for this change into this file: https://github.com/halcyon-tech/vscode-ibmi/blob/master/src/testing/ileErrors.ts

**Comment by angelorpa** [comment]
> @angelorpa Before we can merge this, we need you to pull in the latest changes from `master` and add a test for this change into this file: https://github.com/halcyon-tech/vscode-ibmi/blob/master/src/testing/ileErrors.ts

Hi, I did pulled the latest changes and add two cases for handler errors. Please check it, and let me know if something else is needed.

**Review by worksofliam — APPROVED** [review]
Tests are excellent! Thank you! This is a great addition.

**Pull request merged**